selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/git/GitRevertCommit.java

```diff
@@ -0,0 +1,101 @@
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+/** @author Anatolii Bazko */
+@Singleton
+public class GitRevertCommit {
+  private static final String REVERT_COMMIT_PANEL = "gwt-debug-revert-commit-panel";
+  private static final String REVERT_BUTTON = "git-revert";
+  private static final String CANCEL_BUTTON = "git-revert-cancel";
+
+  private final SeleniumWebDriver seleniumWebDriver;
+  private final Loader loader;
+
+  @Inject
+  public GitRevertCommit(SeleniumWebDriver seleniumWebDriver, Loader loader) {
+    this.seleniumWebDriver = seleniumWebDriver;
+    this.loader = loader;
+    PageFactory.initElements(seleniumWebDriver, this);
+  }
+
+  @FindBy(id = REVERT_COMMIT_PANEL)
+  private WebElement revertPanel;
+
+  @FindBy(id = REVERT_BUTTON)
+  private WebElement revertButton;
+
+  @FindBy(id = CANCEL_BUTTON)
+  private WebElement cancelButton;
+
+  public void waitRevertPanelOpened() {
+    new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(ExpectedConditions.visibilityOf(revertPanel));
+  }
+
+  public void waitRevertPanelClosed() {
+    new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(ExpectedConditions.invisibilityOfElementLocated(By.id(REVERT_COMMIT_PANEL)));
+  }
+
+  public void clickRevertButton() {
+    new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(ExpectedConditions.visibilityOf(revertButton))
+        .click();
+  }
+
+  public String getTopCommitRevision() {
+    loader.waitOnClosed();
+    return new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(ExpectedConditions.visibilityOf(revertPanel))
+        .getText()
+        .split("\n")[1]
+        .replaceAll("\\.", "");
+  }
+
+  public String getTopCommitComment() {
+    loader.waitOnClosed();
+    return new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(ExpectedConditions.visibilityOf(revertPanel))
+        .getText()
+        .split("\n")[4];
+  }
+
+  public String getTopCommitAuthor() {
+    loader.waitOnClosed();
+    return new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
+        .until(ExpectedConditions.visibilityOf(revertPanel))
+        .getText()
+        .split("\n")[3];
+  }
+
+  public void selectRevision(String revision) {
+    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
+        .until(
+            ExpectedConditions.visibilityOfElementLocated(
+                By.xpath("//*[contains(text(),'" + revision + "')]")))
```

Comment:
It looks strange to search on revision number in the whole page. We can potentially find the revision number in the wrong place. IMHO, it's better to restrict search by _Git revert commit panel_.